app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -95,75 +70,32 @@ private void setupItemListener(QuerySnapshot querySnapshots, FirebaseFirestoreEx
-                dateFilterFragment.show(requireActivity().getSupportFragmentManager(), "dates_filter_dialog");
-            } else if (itemId == R.id.filter_by_make) {
-                View makeFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_make, null);
-                MakeFilterFragment makeFilterFragment = new MakeFilterFragment("Modify Make Filter", makeFilterView, this);
-                makeFilterFragment.show(requireActivity().getSupportFragmentManager(), "make_filter_dialog");
-            } else if (itemId == R.id.filter_by_keywords) {
-                View keywordFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_keywords, null);
-                KeywordFilterFragment keywordFilterFragment = new KeywordFilterFragment("Modify Keyword Filter", keywordFilterView, this);
-                keywordFilterFragment.show(requireActivity().getSupportFragmentManager(), "keywords_filter_dialog");
-            } else if (itemId == R.id.filter_by_tags) {
-                View tagFilterView = LayoutInflater.from(getActivity()).inflate(R.layout.fragment_filter_by_tags, null);
-                TagFilterFragment tagFilterFragment = new TagFilterFragment("Modify Tag Filter", tagFilterView, this);
-                tagFilterFragment.show(requireActivity().getSupportFragmentManager(), "tags_filter_dialog");
-            } else {
-                return false;
-            }
-            return true;
-        });
+    protected void unselectAllItems() {
 
-        popupMenu.show();
-    }
+        for (int i=0;i<itemList.size();i++) {
 
-    /**
-     * Called when a filter is applied to the item list. This method adds the filter
-     * to the list of applied filters and triggers the filtering process. If the filter
-     * is already applied, it is removed and added again to re-apply with the new filter value.
-     *
-     * @param filter The filter to be applied to the item list.
-     */
-    @Override
-    public void onFilterApplied(Filter filter) {
-        if (!appliedFilters.add(filter)) {
-            appliedFilters.remove(filter);
-            appliedFilters.add(filter);
-        }
-        applyFilters();
-    }
+            // unselect items
+            itemList.get(i).setSelected(false);
 
-    /**
-     * Applies the list of filters to the item list, resulting in a filtered list of items.
-     * This method iterates through the applied filters, applying each filter in sequence,
-     * and then updates the item adapter with the filtered list of items.
-     */
-    private void applyFilters() {
-        ArrayList<Item> filteredList = new ArrayList<>(itemList);
-        for (Filter filter : appliedFilters) {
-            filteredList = filter.filterList(filteredList);
+            // uncheck selected checkboxes
+            View itemView = itemListView.getChildAt(i);
```

Comment:
Here's my thought, and I will have a corresponding comment at the place where we would also need to change. 

What if we freeze the state of the list when we go into select mode. I.e. once we are in HomeSelectState we disable the listener. Once we delete the selected items we immediately bring up the base (no selection state) and the deletes will update with the snapshot listener. Then we don't even need this